lux-power-distribution-card.js

```diff
@@ -3,99 +3,73 @@ import * as hf from "./html-functions.js";
 import * as constants from "./constants.js";
 
 class LuxPowerDistributionCard extends HTMLElement {
+  constructor() {
+    super();
+    this.attachShadow({ mode: 'open' });
+  }
+
   set hass(hass) {
     this._hass = hass;
 
-    if (!this.card) {
+    if (!this.content) {
       this.createCard();
-      this.bindRefresh(this.card, this._hass, this.config);
-      this.bindHistoryGraph(this.card, this.config);
+      this.bindRefresh(this.card, this._hass, this._config);
+      this.bindHistoryGraph(this.card, this._config);
     }
 
     this.updateCard();
   }
 
   setConfig(config) {
-    if (!this.old_config) {
-      this.old_config = cef.buildConfig(config);
-    }
-    this.config = cef.buildConfig(config);
-    if (this.old_config != this.config) {
-      this.old_config = this.config;
-      this.createCard();
-    }
-
-    // console.log(this.config);
+    this._config = cef.buildConfig(config);
   }
 
   createCard() {
-    if (this.card) {
-      this.card.remove();
-    }
-
-    this.card = document.createElement("ha-card");
-    if (this.config.header) {
-      this.card.header = this.config.header;
+    if (!this._hass || !this._config) {
+      return;
     }
 
-    const content = document.createElement("div");
-    this.card.appendChild(content);
-
-    this.styles = document.createElement("style");
-    this.card.appendChild(this.styles);
-
-    this.appendChild(this.card);
+    const shadowRoot = this.shadowRoot;
 
-    // Set card base
-    content.innerHTML = hf.card_base;
-
-    // Generate Styles
-    this.styles.innerHTML = hf.generateStyles(this.config);
-
-    // Generate Status
-    const inv_info_element = this.card.querySelector("#taskbar-grid");
-    if (inv_info_element) {
-      inv_info_element.innerHTML = hf.generateStatus(this.config);
+    const card = document.createElement('ha-card');
+    if (this._config.title) {
+      const header = document.createElement("h1");
+      header.classList.add('card-header');
+      header.appendChild(document.createTextNode(this._config.title));
+      card.appendChild(header);

```

Comment:
I tried adding the title. This line also need updating in same as the comment above. Changed to `this.card` and now it works. Thanks for adding that feature

Reply:
Sorry about that, I noticed this when I was testing the history graph which didn't work so I adjusted things to use `this.card` but didn't commit it properly. My mistake, i've updated the PR.